### DISCUSSION TITLE
es_host with multiple master nodes ?

### DISCUSSION THREAD

**Discussion by K0WALSKl** [jertel/elastalert2#401] [Q&A] [answered]
Hi,
I have 3 ES master nodes. The [es_host](https://elastalert2.readthedocs.io/en/latest/ruletypes.html?highlight=cluster#es-host) field takes a string but is it possible to add multiple hostnames ? For instance :

`es_host: [hostname1.local, hostname2.local, hostname3.local]`

Thanks in advance :)

**Comment by nsano-rururu** [comment, marked as answer]
It does not correspond. It may be possible to respond by incorporating the contents of the following pull request of yelp/elastalert.

feat: suport for multi es instances
https://github.com/Yelp/elastalert/pull/3109

**Reply by nsano-rururu** [reply]
https://github.com/Yelp/elastalert/issues/2526#issuecomment-551070814
> simple solution is to install a lightweight load balancer (like nginx or haproxy) and redirect your requests to your elasticsearch nodes. Just be sure your index settings are correctly spreading replicas of your shards accross your nodes because if the LB query a elastic node but doesn't contain all the data you needs, you will not see everything if the datas are spreads between the nodes

https://github.com/Yelp/elastalert/issues/1402#issuecomment-339786817
>No, you don't need to install it multiple places, nor do you need to declare all the node names. All you need is direct connectivity to a SINGLE node.

https://github.com/Yelp/elastalert/issues/1364#issuecomment-332835117
>Please use haproxy in front of elasticsearch . This helps to accomplish ha mode for elastalert.

https://github.com/Yelp/elastalert/issues/614#issuecomment-230896756
>I recommend haproxy. It's super simple to set up a reverse proxy for Elasticsearch to do load balancing.

**Reply by K0WALSKl** [reply]
I did not know that was possible :o
I will definitely try this, thank you very much!!
